node/src/components/consensus/highway_core/highway.rs

```diff
@@ -583,12 +583,25 @@ impl<C: Context> Highway<C> {
     }
 
     /// Checks whether the unit was created by a doppelganger.
-    pub(crate) fn is_doppelganger(&self, swunit: &SignedWireUnit<C>) -> bool {
-        let creator = swunit.wire_unit.creator;
-        self.active_validator
-            .as_ref()
-            .map(|av| av.vidx == creator)
-            .unwrap_or(false)
+    pub(crate) fn is_our_own(&self, vertex: &Vertex<C>) -> bool {
```

Comment:
Now that you _do_ check that it's not in the state, I actually think it should be called `is_doppelgangers_vertex` instead of `is_our_own`. (Because I'd expect `is_our_own` to be true for _all_ vertices we created.)

Reply:
fxed.